packages/components/src/internal/components/user/UsersGridPanel.tsx

```diff
@@ -173,20 +173,24 @@ export class UsersGridPanelImpl extends PureComponent<Props, State> {
         this.closeDialog();
         this.onRowSelectionChange(this.getUsersModel(), undefined, false); // clear selected user details
         this.props.onCreateComplete(response, roles);
-        this.reloadUsersModel();
+        this.props.actions.onModelChange(this.getUsersModelId(), { changeType: ChangeType.add });
     };
 
     reloadUsersModel(): void {
         this.props.actions.loadModel(this.getUsersModelId(), true, true);
     }
 
-    onUsersStateChangeComplete = (response: any, resetSelection = true): void => {
+    onUsersStateChangeComplete = (response: any, isDelete: boolean = false): void => {
         this.closeDialog();
-        if (resetSelection) {
-            this.onRowSelectionChange(this.getUsersModel(), undefined, false); // clear selected user details
-        }
+        this.updateSelectedUserId(undefined); // clear selected user details
```

Comment:
this will clear the selected user for all cases of `onUsersStateChangeComplete` now, wouldn't we want to only do that for the delete case and not the other cases (like the activate change modal for example)?

Reply:
Yes, I will correct that